File: python/chcube.py
t = input ()
for i in range(t):
	str = raw_input ()
	elem = str.split(" ")
	arr = [0, 0, 0 ,0, 0, 0]
	for i in elem:
		if (i == 'black'):
			arr[0] += 1
		elif (i == 'blue'):
			arr[1] += 1 
		elif (i == 'red') :
			arr[2] += 1 
		elif (i == 'green') :
			arr[3] += 1 
		elif (i == 'yellow'):
			arr[4] += 1 
		elif (i == 'orange'):
			arr[5] += 1 
	flag = 0 
	for i in  arr : 
		if (arr[i] >= 3):
			flag = 1 
	if (flag == 1 ):
		print 'YES'	
	else :
		print 'NO'

File: C/chcube.c
#include<stdio.h>
#include<string.h>

int main ()
{
	int t , arr[8] , i  , j  , flag  ; 
	char str[20] ; 
	scanf ("%d",&t ); 
	while (t--) 
	{
		for (i=0 ; i<6 ; i++){
			arr[i] = 0 ; 
		}
		flag = 0 ; 
		for (i=0 ; i<6 ; i++){
			scanf ("%s",str);
			if (str == "black")
				arr[0] ++ ; 
			else if (str == "blue") 
				arr[1] ++ ; 
			else  if (str == "red")
				arr[2] ++ ; 
			else if (str == "green") 
				arr[3] ++ ; 
			else  if (str == "yellow")
				arr[4] ++ ; 
			else if (str == "orange")
				arr[5] ++ ; 
		}

		for(i = 0 ; i<6 ; i++){
			if (arr[i] >=3)
				flag = 1; 
		}
		if (flag == 1 )
			printf ("YES\n") ; 
		else 
			printf ("NO\n"); 
	}
return 0 ; 

}
